lib/league_stats.rb

```diff
@@ -32,37 +32,61 @@ def lowest_defense
   end
 
   def highest_scoring_when_away
-    @teams.max_by do |team|
-      team.away_average_goals
+    teams = @teams.reject do |team|
+      team.average_goals(team.away).to_s == 'NaN'
+    end
+
+    teams.max_by do |team|
+      team.average_goals(team.away)
     end.team_name
   end
 
   def highest_scoring_when_home
-    @teams.max_by do |team|
-      team.home_average_goals
+    teams = @teams.reject do |team|
+      team.average_goals(team.home).to_s == 'NaN'
+    end
+
+    teams.max_by do |team|
+      team.average_goals(team.home)
     end.team_name
   end
 
   def lowest_scoring_when_away
-    @teams.min_by do |team|
-      team.away_average_goals
+    teams = @teams.reject do |team|
+      team.average_goals(team.away).to_s == 'NaN'
+    end
+
+    teams.min_by do |team|
+      team.average_goals(team.away)
     end.team_name
   end
 
   def lowest_scoring_when_home
-    @teams.min_by do |team|
-      team.home_average_goals
+    teams = @teams.reject do |team|
+      team.average_goals(team.home).to_s == 'NaN'
+    end
+
+    teams.min_by do |team|
+      team.average_goals(team.home)
     end.team_name
   end
 
   def highest_win_percentage
-    @teams.max_by do |team|
-      team.win_percentage
+    teams = @teams.reject do |team|
+      team.win_percentage(team.total).to_s == 'NaN'
+    end
+
+    teams.max_by do |team|
+      team.win_percentage(team.total)
     end.team_name
   end
 
   def highest_fans_rating
-    @teams.max_by do |team|
+    teams = @teams.reject do |team|
+      team.win_percentage(team.home).to_s == 'NaN' || team.win_percentage(team.away).to_s == 'NaN'
```

Comment:
Shouldn't these just be "team.win_percentage(:home).to_s"?  Same with :away

Reply:
Actually I see what you were doing. Not just passing a type but the actual hash. Believe it will need to be "team.history[:home/:away]" now.